lib/ansible/modules/net_tools/nios/nios_host_record.py

```diff
@@ -252,7 +278,7 @@ def main():
     ''' Main entry point for module execution
     '''
     ipv4addr_spec = dict(
-        ipv4addr=dict(required=True, aliases=['address'], ib_req=True),
+        ipv4addr=dict(required=True, aliases=['address', 'address_add', 'address_remove'], ib_req=True),
```

Comment:
aliases in options are meant to have the same meaning for example `commands` and `lines` in `ios_config` modules. In this case, if I understand correctly the functionality changes with the aliases used.
If the intention is to add/remove address can't it done using state options.
```
- name: remove ip to host record
  nios_host_record:
    name: host.ansible.com
    ipv4:
      - address: 192.168.10.1
    state: absent
    provider: "{{ provider }}"
  connection: local
```
Since ipv4 address value is mentioned `state: absent` can mean remove ipv4 address only and the entire host record.